substrate/frame/support/src/genesis_builder_helper.rs

```diff
@@ -55,3 +55,82 @@ where
 		preset_for_name,
 	)
 }
+
+/// Creates a `RuntimeGenesisConfig` JSON patch.
+///
+/// This macro creates a default `RuntimeGenesisConfig` initializing provided fields with given
+/// values, serialize it to JSON blob, and retain only the specified fields.
+///
+/// This macro helps to prevents errors that could occur from manually creating JSON objects, such
+/// as typos or discrepancies caused by future changes to the `RuntimeGenesisConfig` structure. By
+/// using the actual struct, it ensures that the JSON generated is valid and up-to-date.
+///
+/// This macro assumes that `serde(rename_all = "camelCase")` attribute is used for
+/// RuntimeGenesisConfig, what should be the case for frame-based runtimes.
+///
+/// # Example
+///
+/// ```rust
+/// use frame_support::runtime_genesis_config_json;
+/// #[derive(Default, serde::Serialize, serde::Deserialize)]
+/// #[serde(rename_all = "camelCase")]
+/// struct RuntimeGenesisConfig {
+///     a_field: u32,
+///     b_field: u32,
+///     c_field: u32,
+/// }
+/// assert_eq!(
+///     runtime_genesis_config_json! ({a_field:31, b_field:127}),
+///     serde_json::json!({"aField":31, "bField":127})
+/// );
+/// assert_eq!(
+///     runtime_genesis_config_json! ({a_field:31}),
+///     serde_json::json!({"aField":31})
+/// );
+/// ```
+#[macro_export]
+macro_rules! runtime_genesis_config_json {
+	({ $( $key:ident : $value:expr ),* $(,)? }) => {{
+        let config = RuntimeGenesisConfig {
+            $( $key : $value, )*
+            ..Default::default()
+        };
+
+		#[inline]
+        fn compare_keys(
```

Comment:
We could also use [`convert_case`](https://docs.rs/convert_case/0.6.0/convert_case/) crate. But this probably would be unnecessary overhead.

Reply:
Thinking about it, does it handle all the weird edge cases? In theory, the pallet authors can name their field names whatever they like. Like `A___B`, `ABC_ABC` or whatever.  Maybe better to use the crate after all.

Reply:
Not sure if this is needed. The names in `RuntimeGenesisConfig` are also [generated](https://github.com/paritytech/polkadot-sdk/blob/f6d08e637694729df7cddb166954ee90c2da3ce1/substrate/frame/support/procedural/src/construct_runtime/expand/config.rs#L41) by the `construct_runtime` macro. I assume no weird naming will happen there - don't know if you can name your pallet with `_` within the name.


Anyway, I see that [Inflector](https://docs.rs/Inflector/latest/inflector/cases/camelcase/index.html) is already used in our codebase so I will rework this.
If something will pop-up we can fix it later.

Reply:
The other question is how serde_json is generating camel_case. We should use exactly the same implementation if we want to be 100% accurate.

Reply:
One more thought: `Inflector` is used in `procedural` and not in runtime.

Reply:
sadly `Inflector` has no support for `nostd`.

Reply:
> Thinking about it, does it handle all the weird edge cases? In theory, the pallet authors can name their field names whatever they like. Like `A___B`, `ABC_ABC` or whatever. Maybe better to use the crate after all.

This is done in 6db4b41